azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -0,0 +1,144 @@
+from azurelinuxagent.common.event import WALAEventOperation, add_event
+from azurelinuxagent.common import conf
+import platform
+
+# Define support matrix for Regorus and policy engine feature.
+# Dict in the format: { distro:min_supported_version }
+POLICY_SUPPORTED_DISTROS_MIN_VERSIONS = {
+    'ubuntu': DistroVersion('16.04'),
+    'mariner': DistroVersion('2'),
+    'azurelinux': DistroVersion('3')
+}
+# TODO: add 'arm64', 'aarch64' here once support is enabled for ARM64
+POLICY_SUPPORTED_ARCHITECTURE = ['x86_64', 'amd64']
+
+
+class PolicyEngine(object):
+    """
+    Implements policy engine API. Class will always be initialized, but if the Regorus import fails,
+    all methods will be no-ops.
+
+    If any errors are thrown in regorus.py, they will be caught and handled here. add_policy, add_data,
+    and set_input will be no-ops, eval_query will return an empty dict
+    """
+    def __init__(self, policy_file, data_file):
+        # TODO: package default policy and data file with agent, set paths as default constructor parameters
+        self._policy_engine_enabled = False
+        self._engine = None
+        if not self.is_policy_enforcement_enabled():
+            self.log_policy(msg="Policy enforcement is not enabled.")
+            return  # policy_engine_enabled is not set to True
+
+        if not self.is_policy_enforcement_supported():
+            # TODO: Update this error message when conf flag is removed post private preview.
+            msg = "Attempted to enable policy enforcement, but feature is not supported on this platform."
+            self.log_policy(msg=msg)
+            raise Exception(msg)
+
+        try:
+            import azurelinuxagent.ga.policy.regorus as regorus
+            self._engine = regorus.Engine()
+            self._engine.add_policy(policy_file)
+            self._engine.add_data(data_file)
+            self._policy_engine_enabled = True
+
+        except Exception as ex:
+            msg = "Failed to initialize Regorus policy engine: '{0}'".format(ex)
+            self.log_policy(msg=msg)
+            raise Exception(msg)
+
+    @classmethod
+    def log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_event=True):
+        """
+        Log information to console and telemetry.
+        """
+        if is_success:
+            logger.info("[Policy] " + msg)
+        else:
+            logger.error("[Policy] " + msg)
+        if send_event:
+            add_event(op=op, message=msg, is_success=is_success)
+
+    @staticmethod
+    def is_policy_enforcement_enabled():
+        """Check whether user has opted into policy enforcement feature"""
+        # TODO: Remove conf flag post private preview and add other checks here.
+        return conf.get_extension_policy_enabled()
+
+    @staticmethod
+    def is_policy_enforcement_supported():
+        """Check that both platform architecture and distro/version are supported."""
+        arch = platform.machine().lower()
```

Comment:
We have an osutil function for this: osutil.get_vm_arch()

Reply:
updated to use osutil